backend/src/main/java/com/bakdata/conquery/models/messages/namespaces/specific/UpdateElementMatchingStats.java

```diff
@@ -26,27 +29,40 @@ public class UpdateElementMatchingStats extends NamespaceMessage {
 	private final WorkerId source;
 
 	@ToString.Exclude
-	@NsIdRefKeys
-	private final Map<ConceptElement<?>, MatchingStats.Entry> values;
+	private final Map<ConceptElementId<?>, MatchingStats.Entry> values;
 
 	@Override
 	public void react(DistributedNamespace context) throws Exception {
-		for (Entry<ConceptElement<?>, MatchingStats.Entry> entry : values.entrySet()) {
+		// We collect the concepts outside the loop to update the storage afterward
+		Map<ConceptId, Concept<?>> conceptsToUpdate = new HashMap<>();
+
+		for (Entry<ConceptElementId<?>, MatchingStats.Entry> entry : values.entrySet()) {
 			try {
-				final ConceptElement<?> target = entry.getKey();
+				ConceptElementId<?> element = entry.getKey();
+				ConceptId conceptId = element.findConcept();
+
+				// mapping function cannot use Id::resolve here yet, somehow the nsIdResolver is not set because it
```

Comment:
das klingt nach einem Bug in dem Deserliazier? Vlt brauchst du einen dedizierten Key Deserializer

Reply:
Das hatte ich noch nicht ganz verstanden. Ich war davon ausgegangen, dass der Normale Deserialiser genutzt wird, wenn kein KeyDeserializer genutzt wird. Aber es ist wohl am besten mit einem Test herauszufinden.